File: 백준/Gold/1806. 부분합/부분합.py
import sys
input = sys.stdin.readline
n, s = map(int, input().split())
numList = list(map(int, input().split()))

if sum(numList) < s:
    print(0)
    exit()

end = 0
tmp = 0
answer = n
for start in range(n):
    while tmp < s and end < n:
        tmp += numList[end]
        end += 1
    
    if tmp >= s:
        answer = min(answer, end - start)
    tmp -= numList[start]

print(answer)

File: 백준/Gold/1806. 부분합/부분합.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let lines = [];

rl.on("line", (input) => {
  lines.push(input.trim());
});

rl.on("close", () => {
  const [n, s] = lines[0].split(" ").map(Number);
  const arr = lines[1].split(" ").map(Number);

  let start = 0;
  let end = 0;
  let sum = 0;
  let answer = Infinity;
  while (end < n) {
    sum += arr[end];

    while (sum >= s) {
      answer = Math.min(answer, end - start + 1);
      sum -= arr[start];
      start++;
    }
    end++;
  }
  console.log(answer === Infinity ? 0 : answer);
});
